tests/js/server/resilience/moving-shards-with-arangosearch-view-cluster.js

```diff
@@ -444,111 +571,137 @@ function MovingShardsWithViewSuite (options) {
+  }
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief checks if the collection is in sync with the plan
+////////////////////////////////////////////////////////////////////////////////
 
+  function planEqualCurrent(collection) {
     global.ArangoClusterInfo.flush();
+    const shardDist = internal.getCollectionShardDistribution(collection._id);
+    const Plan = shardDist[collection.name()].Plan;
+    const Current = shardDist[collection.name()].Current;
 
-    var cinfo = global.ArangoClusterInfo.getCollectionInfo(
-        database, collection.name());
-    for(var shard in cinfo.shards) {
-      getDBServers().forEach(serverId => {
-        var dbServerEndpoint = global.ArangoClusterInfo.getServerEndpoint(serverId);
-        var url = endpointToURL(dbServerEndpoint);
-
-        var res;
-        try {
-          var envelope =
-              { method: "GET", url: url + "/_api/view" };
-          res = request(envelope);
-        } catch (err) {
-          console.error(
-            "Exception for GET /_api/view:", err.stack);
-          return {};
-        }
-        var body = res.body;
-        if (typeof body === "string") {
-          var views = JSON.parse(body).result;
-        }
-        views.forEach(v => {
-          var res;
-          try {
-            var envelope =
-                { method: "GET", url: url + "/_api/view/" + v.name + "/properties" };
-            res = request(envelope);
-          } catch (err) {
-            console.error(
-              "Exception for GET /_api/view/" + v.name + "/properties:", err.stack);
-            return {};
-          }
-          var body = res.body;
-          if (typeof body === "string") {
-            var links = JSON.parse(body).links;
-            if (links !== undefined
-                && links.hasOwnProperty(shard)
-                && links.shard !== {}) {
-              sViewServers.push(serverId);
-            }
-          }
-        });
-      });
-    }
+    return _.isObject(Plan) && _.isEqual(Plan, Current);
+  }
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief waits for the collection to come in sync with the plan
+////////////////////////////////////////////////////////////////////////////////
+
+  function waitForPlanEqualCurrent(collection) {
+    return waitUntilWithTimeout(
+      () => planEqualCurrent(collection),
+      120, 1.0,
+      `Collection "${collection}" failed to get plan in sync after 120 sec`
+    );
+  }
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief Waits for views to be available on all dbservers they should be
+/// created on, including followers. This also asserts the view is not available
+/// on servers it should not be.
+////////////////////////////////////////////////////////////////////////////////
+
+  function waitForViewsToGetInSync() {
+
```

Comment:
Will this method check that IDs of DBServers for collections and sharded views are the same or only number of servers is checked (i.e. `planServers, server => currentServers.has(server)` is enough)?

Reply:
It checks that every DBServer that is in the plan is also in current (for each shard), and that there are as many DBServers in the plan and in current. Thus it checks that the DBServers are the same, up to order.
I'll add comments to make that clear.

Reply:
And with current I mean, available via the view api of the particular DBServer.